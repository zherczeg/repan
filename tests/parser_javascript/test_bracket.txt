# Testing bracket types.

/(ab)(?:cd)(?=ef)(?!gh)(?<=ij)(?<!kl)/J
    print

/(?|ab)/J
    print

/(?<name>ab)cd\k<name>/J
    print

/(?<Aab\u0063>def)\k<\u0041abc>/J
    print

/(?<>def)/J
    print

/\k<#>/J
    print

/(?<\u0030>def)/J
    print

/(?<a\u30>def)/J
    print

/(?<\u{2dde}>)(?<\u{2e2f}>)/16uJ
    print

/(?<\u{2d000}>)(?<a\u{2e2f}>)/16uJ
    print

# Global modifiers

/.[^]/J
    print

/.[^]/Js
    print
