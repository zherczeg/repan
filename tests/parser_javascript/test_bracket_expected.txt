# Testing bracket types.

/(ab)(?:cd)(?=ef)(?!gh)(?<=ij)(?<!kl)/J
    print
    /(ab)(?:cd)(?=ef)(?!gh)(?<=ij)(?<!kl)/

/(?|ab)/J
    print
    Pattern parse failed: Invalid subpattern type
        /[ HERE ]>>(?|ab)/

/(?<name>ab)cd\k<name>/J
    print
    /(?<name>ab)cd\g{name}/

/(?<Aab\u0063>def)\k<\u0041abc>/J
    print
    /(?<Aabc>def)\g{Aabc}/

/(?<>def)/J
    print
    Pattern parse failed: A valid capturing group name is expected
        /(?<[ HERE ]>>>def)/

/\k<#>/J
    print
    Pattern parse failed: A valid capturing group name is expected
        /\k<[ HERE ]>>#>/

/(?<\u0030>def)/J
    print
    Pattern parse failed: A valid capturing group name is expected
        /(?<[ HERE ]>>\u0030>def)/

/(?<a\u30>def)/J
    print
    Pattern parse failed: Four hexadecimal digits are required after \u
        /(?<a[ HERE ]>>\u30>def)/

/(?<\u{2dde}>)(?<\u{2e2f}>)/16uJ
    print
    Pattern parse failed: A valid capturing group name is expected
        /(?<\u{2dde}>)(?<[ HERE ]>>\u{2e2f}>)/

/(?<\u{2d000}>)(?<a\u{2e2f}>)/16uJ
    print
    Pattern parse failed: A character allowed in capturing group names is expected
        /(?<\u{2d000}>)(?<a[ HERE ]>>\u{2e2f}>)/
